wrapper/src/libvmaf.h

```diff
@@ -32,92 +34,140 @@
+    VMAF_MODEL_SETTING_ENABLE_TRANSFORM                = (1 << 1),
+    VMAF_MODEL_SETTING_DISABLE_CLIP                    = (1 << 2),
+    VMAF_MODEL_SETTING_ENABLE_CONF_INTERVAL            = (1 << 3),
 };
 
-class VmafQualityRunnerFactory {
-public:
-    static std::unique_ptr<IVmafQualityRunner> 
-        createVmafQualityRunner(const char *model_path, bool enable_conf_interval);
-};
+/** Thin VMAF model wrapper.
+  * Assumes the existence of the model in a filesystem path.
+  * Includes a model setting for disabling clip, score transformation etc. */
+typedef struct {
+    char *name;
+    char *path;
+    int vmaf_model_setting;
+} VmafModel;
+
+typedef struct {
+
+    unsigned int width;
+    unsigned int height;
+    unsigned int num_models;
+    unsigned int default_model_ind; // useful to tell between the default model and additional models (if any)
+
+    char *log_path;
+    int vmaf_feature_mode_setting;
+
+    VmafModel vmaf_model[MAX_NUM_VMAF_MODELS];
+    VmafFeatureCalculationSetting vmaf_feature_calculation_setting;
+
+    enum VmafPixelFormat pix_fmt;
+    enum VmafLogFmt log_fmt;
+    enum VmafPoolingMethod pool_method;
 
+} VmafSettings;
+
+/** Computes VMAF using a read_vmaf_picture callback,
+  * a user-defined struct and a settings struct.
+  * A single VMAF value is returned and all calculated
+  * features and additional predictions are written into a log file. */
+int compute_vmaf(VmafOutput *vmaf_output_ptr,
+                 int (*read_vmaf_picture)(VmafPicture *ref_vmaf_pict, VmafPicture *dis_vmaf_pict, float *temp_data, void *user_data),
+				 void *user_data, VmafSettings *vmafSettings);
+
+/** Populates additional models to be used for prediction.
+  * A maximum of
+  * The function returns the number of additional models as an integer (0 for a NULL ptr).
+  * The first model is the model corresponding to model_path.
+  * The additional models are passed in using the additional_model_paths string.
+  * This string contains model-specific parameters for each additional model (path, disable_clip, etc.)
+  * This string should be formatted as an escaped json-like string with the following format:
+  * {\"<name>\":{\"model_path\"\:\"<path>\"}}, e.g.:
+  * {\"another_model\":{\"model_path\"\:\"some/path/to/some/model/another_model.pkl\"}}
+  * The fields <name> and <path> are required and optional parameters can be used by adding an escaped comma, e.g.:
+  * {\"<name>\":{\"model_path\"\:\"<path>\"\,\"disable_clip\"\:\"1\"}} --> disable clipping of predictions
+  * {\"<name>\":{\"model_path\"\:\"<path>\"\,\"enable_transform\"\:\"1\"}} --> use transformation for the predictions (e.g. phone model transformation)
+  * {\"<name>\":{\"model_path\"\:\"<path>\"\,\"enable_conf_interval\"\:\"1\"}} --> calculate confidence interval (CI) for the predictions (the model should support CIs)
+  * {\"<name>\":{\"model_path\"\:\"<path>\"\,\"disable_clip\"\:\"0\"\,\"enable_transform\"\:\"1\"}} --> enable clipping and transform scores
```

Comment:
If someone is using this as a C library they need to allocate a` char*` with enough space to hold a properly escaped and formatted JSON string?

Could this design be better?

Reply:
In that case one can create a VmafModel similar to how run_wrapper inits the first model.

Reply:
vmafSettings->vmaf_model[0].name = "vmaf";
    vmafSettings->vmaf_model[0].path = model_path;
    vmafSettings->vmaf_model[0].vmaf_model_setting = vmaf_model_setting;